Content.Server/Ame/EntitySystems/AmeFuelSystem.cs

```diff
@@ -27,4 +66,20 @@ private void OnFuelExamined(EntityUid uid, AmeFuelContainerComponent comp, Exami
             ("amount", comp.FuelAmount),
             ("capacity", comp.FuelCapacity)));
     }
+
+    /// <summary>
+    /// Makes the fuel jar start leaking antimatter.
+    /// </summary>
+    public void StartLeaking(EntityUid uid, AmeFuelContainerComponent? comp = null)
+    {
+        if (!Resolve(uid, ref comp) || comp.Leaking)
+            return;
+
+        if (comp.FuelAmount < 1 || !HasComp<ExplosiveComponent>(uid))
+            return;
+
+        // don't instantly explode, give the user a little time to react
+        comp.Leaking = true;
```

Comment:
would rather it added an `AmeFuelLeakingComponent` or something and queried over that so it isnt querying over every jar for no reason

Reply:
@mirrorcult while im at to make the destruction behaviour generic, how would i addcomp a component specified in yaml so its not so extremely specific

Reply:
Some sort of 'fuse' component? Delay and explosion prototype?

Reply:
or simply re-use some explosive components? from c-4 or explosive pen